android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -495,6 +504,13 @@ public static void initialize(final Context context, KeyboardType keyboardType)
     migrateCloudKeyboards(appContext);
 
     CloudDownloadMgr.getInstance().initialize(appContext);
+
+    KeyboardHeight_Context_Portrait_Default = calculateDefaultKeyboardHeight(context, Configuration.ORIENTATION_PORTRAIT); 
+    KeyboardHeight_Context_Landscape_Default = calculateDefaultKeyboardHeight(context, Configuration.ORIENTATION_LANDSCAPE);
+    SharedPreferences prefs = context.getSharedPreferences(context.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
+    KeyboardHeight_Context_Portrait_Current = prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, KMManager.KeyboardHeight_Context_Portrait_Default);
+    KeyboardHeight_Context_Landscape_Current = prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, KMManager.KeyboardHeight_Context_Landscape_Default);
```

Comment:
I think we can only get resources when not test mode (like inside of calculateDefaultKeyboardHeight

Maybe wrap this block in 
```
if (!testMode()) {

}
```

Reply:
No, that's going in the wrong direction... We should stop using `context.getString(R.string.kma_prefs_name)` and instead just use a constant.

Let's introduce the constant here and then fixup across the board in the rest of the codebase.

Reply:
I fixed it here with a constant: "KMEngine_PrefsKey".